src/zenml/integrations/azure/service_connectors/azure_service_connector.py

```diff
@@ -0,0 +1,1777 @@
+            if resource_type not in [AZURE_RESOURCE_TYPE, None]
+            else None,
+            expiration_seconds=expiration_seconds,
+            expires_at=expires_at,
+            config=auth_config,
+        )
+
+    def _list_blob_containers(
+        self, credential: TokenCredential, container_name: Optional[str] = None
+    ) -> Dict[str, str]:
+        """Get the list of blob storage containers that the connector can access.
+
+        Args:
+            credential: The Azure credential to use to access the blob storage
+                containers.
+            container_name: The name of the blob container to get. If omitted,
+                all accessible blob containers are returned.
+
+        Returns:
+            The list of blob storage containers that the connector can access
+            as a dictionary mapping container names to their corresponding
+            storage account names. If the `container_name` argument was
+            specified, the dictionary will contain a single entry.
+
+        Raises:
+            AuthorizationException: If the connector does not have access to
+                access the target blob storage containers.
+        """
+        subscription_id, _ = self.subscription
+        # Azure blob storage containers are scoped to a storage account. We
+        # need to figure out which storage accounts the connector can access
+        # and then list the containers in each of them. If a container name
+        # is provided, we only need to find the storage account that contains
+        # it.
+
+        storage_accounts: List[str] = []
+        if self.config.storage_account:
+            storage_accounts = [self.config.storage_account]
+        else:
+            try:
+                storage_client = StorageManagementClient(
+                    credential, subscription_id
+                )
+                accounts = list(storage_client.storage_accounts.list())
+            except AzureError as e:
+                raise AuthorizationException(
+                    f"failed to list available Azure storage accounts. Please "
+                    f"check that the credentials have permissions to list "
+                    f"storage accounts or consider configuring a storage "
+                    f"account in the connector: {e}"
+                ) from e
+
+            if not accounts:
+                raise AuthorizationException(
+                    "no storage accounts were found. Please check that the "
+                    "credentials have permissions to access one or more "
+                    "storage accounts."
+                )
+
+            if self.config.resource_group:
+                # Keep only the storage accounts in the specified resource
+                # group.
+                accounts = [
+                    account
+                    for account in accounts
+                    if account.id.split("/")[4] == self.config.resource_group
```

Comment:
I've seen this `split("/")[4]` in a few places now, should we extract it in some function in case it needs to be updated in the future?